utils/crypto_utils.h

```diff
@@ -16,19 +16,102 @@ extern "C" {
 #endif
 
 #define AES_BLOCK_SIZE 16
+#define AES_CBC_KEY_SIZE AES_BLOCK_SIZE * 2
 #define MAXLINE 1024
 #define IMSI_LEN 15
 
+/**
+ * @brief Gets device IMSI or physical ID
+ *
+ * @param[out] device_id Device id output
+ *
+ * @return
+ * - zero on success
+ * - -1 on error
+ */
 int get_device_id(const char *device_id);
+
+/**
+ * @brief Gets aes key from hashchain
+ *
+ * @param[out] key Device key output
+ *
+ * @return
+ * - zero on success
+ * - -1 on error
+ */
 int get_aes_key(const uint8_t *key);
-int encrypt(unsigned char *plaintext, int plaintext_len, unsigned char *ciphertext, int ciphertext_len, uint8_t iv[16],
-            uint8_t key[AES_BLOCK_SIZE * 2], uint8_t device_id[IMSI_LEN + 1]);
-int decrypt(unsigned char *ciphertext, int ciphertext_len, unsigned char *plaintext, int plaintext_len, uint8_t iv[16],
-            uint8_t key[AES_BLOCK_SIZE * 2]);
-int aes_encrypt(const unsigned char *plaintext, int plaintext_len, const unsigned char *key, unsigned int keybits,
-                unsigned char iv[16], unsigned char *ciphertext, int ciphertext_len);
-int aes_decrypt(const unsigned char *ciphertext, int ciphertext_len, const unsigned char *key, unsigned int keybits,
-                unsigned char iv[16], unsigned char *plaintext, int plaintext_len);
+
+/**
+ * @brief Encrypt plaintext to ciphertext
+ *
+ * @param[in] plaintext The text to be encrypted
+ * @param[in] plaintext_len Plaintext length
+ * @param[out] ciphertext Ciphrtext
+ * @param[in] ciphertext_len Ciphertext length
+ * @param[in, out] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[in] key[AES_CBC_KEY_SIZE] Encryption key
+ * @param[in] device_id[IMSI_LEN + 1] Device id
+ *
+ * @return
+ * - Ciphertext text new length on success
+ * - 0 on error
+ */
+int encrypt(const char *plaintext, int plaintext_len, char *ciphertext, int ciphertext_len, uint8_t iv[AES_BLOCK_SIZE],
+            uint8_t key[AES_CBC_KEY_SIZE], uint8_t device_id[IMSI_LEN + 1]);
+
+/**
+ * @brief Decrypt ciphertext to plaintext
+ *
+ * @param[in] ciphertext Ciphertext
+ * @param[in] ciphertext_len Ciphertext length
+ * @param[out] plaintext Plaintext
+ * @param[in] plaintext_len Plaintext length
+ * @param[in] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[in] key[AES_CBC_KEY_SIZE] Decryption key
+ *
+ * @return
+ * - zero on success
+ * - -1 on error
+ */
+int decrypt(const char *ciphertext, int ciphertext_len, char *plaintext, int plaintext_len, uint8_t iv[AES_BLOCK_SIZE],
+            uint8_t key[AES_CBC_KEY_SIZE]);
+
+/**
+ * @brief Implementation of aes encrypt
+ *
+ * @param[in] plaintext Plaintext
+ * @param[in] plaintext_len Plaintext length
+ * @param[in] key Encryption key
+ * @param[in] keybits Length of key
+ * @param[in, out] iv[AES_BLOCK_SIZE] Initialization vector
+ * @param[out] ciphertext Ciphertext
+ * @param[in] ciphertext_len Ciphertext length
```

Comment:
Don't put a input parameter after a output parameter. The order will mislead people

Reply:
New issue #54 for this.